### DISCUSSION TITLE
leave comment on file names

### DISCUSSION THREAD

**Discussion by caoilte** [community/community#13013] [Pull Requests] [answered]
I often want to leave pull request comments requesting a file name change. AFAICT there is no direct way to do this and so I always end up leaving a comment on the first line of the file instead. It would be a lot cleaner if I could leave a comment on the whole file (especially if i could do it as a code suggestion).

**Comment by cschneid** [comment]
Similarly, sometimes a discussion is about the entire file, not any particular lines inside of it.

We had the case recently where a whole file looked pretty ugly, but for fairly good reasons. On a PR comment, i'd like to reiterate those reasons, and be able to have a short discussion thread about that. 

But I'm limited to commenting on Line #1, and while it's often a code smell, `import React` is not what I'm talking about 😄.

Similarly, there's other parts of the PR to really review, and leaving just a top-level description on the PR doesn't point people at this important spot for discussion.

**Comment by kilaka** [comment]
Does it sound reasonable to comment on the first line until we have such a feature?
This is out of the first line's context but good enough for now?

**Comment by yo1995** [comment]
Sometime a change is file rename only, which makes it impossible to leave comment on the first line of the file. It would great if there is a way to suggest a change to the file name/path.

**Comment by ptrkstr** [comment]
Another use case I have for this is sometimes a PR contains a binary file change, for example a snapshot test having a reference image updated. I'd like the ability to comment on the binary file change. I currently cannot as there are no source code lines to start the comment.

**Comment by aditya81070** [comment]
Another use case, sometimes I want to comment on the location of the file so I have to comment on the first line and tell that `code looks good on this line but can we move this file to folder x`.

**Comment by yo1995** [comment, marked as answer]
I noticed GitHub added this feature (actually partially, currently can only comment, not suggest change) today! https://github.com/community/community/discussions/49954

**Comment by manokjeet-sketch** [comment]
I used to think talent mattered most, but regular practice always gave me better results in the long run.
A good reminder is to keep building small projects consistently, and [basic programming language commands pdf](https://basiccodingconcepts.net/) has been a helpful resource for reinforcing the basics.

**Reply by v-ember** [reply]
Hi @manokjeet-sketch, thanks for sharing with the community! 

Unfortunately, we currently do not allow self-promotion, advertising, or solicitation in Community Discussions. We want to make sure there is space for users to ask questions without overwhelming them with other conversations. Thank you for helping us maintain a productive and tidy community for all our members.
